packages/fxa-shared/nestjs/gql/gql-whitelist.ts

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { readFileSync } from 'fs';
+import { Request, Response, NextFunction } from 'express';
+
+/**
+ * Configuration options for GqlGuard
+ */
+export type Config = {
+  /** A list of json files holding queries extracted with the persistgraphql util. */
+  whitelist: string[];
+  /** Toggles gaurd on / off. */
+  enabled: boolean;
+};
+
+/**
+ * Guards against the execution of unsanctioned gql queries. A list of allowed queries can be populated by running.
+ * `yarn gql:whitelist`.
+ *
+ * During development it is probably desirable to disable this!
+ */
+export class GqlWhitelist {
+  public readonly valid: Array<Record<string, number>>;
+  public readonly enabled: boolean;
+
+  constructor(config: Config) {
+    this.valid = config.whitelist.map((x) => {
+      const raw = readFileSync(x).toString();
+      try {
+        return JSON.parse(raw);
+      } catch (err) {
+        console.warn('Failed to parse gql whitelist', raw);
+      }
+    });
+    this.enabled = config.enabled;
```

Comment:
I'd expect that we don't need to parse the whitelist if the feature isn't enable.